go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1964,6 +1972,55 @@ func (call *builtinYearWeek) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Int64, Col: collationNumeric, Flag: arg.Flag | flagNullable}, nil
 }
 
+func periodAdd(period, months int64) (*evalInt64, error) {
+	if !datetime.ValidatePeriod(period) {
+		return nil, vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.WrongArguments, "Incorrect arguments to period_add")
+	}
+	return newEvalInt64(int64(datetime.MonthsToPeriod(datetime.PeriodToMonths(uint64(period)) + uint64(months)))), nil
```

Comment:
this seems to be the only use of `MonthsToPeriod` and `PeriodToMonths`. Wouldn't it make sense to have it accept int64 in, and return int64 values? that way we wouldn't have to cast them here

Reply:
Yeah, not a huge issue but let's make it consistent so we don't cast here. @beingnoble03 Any reason it must be using an unsigned type?

Reply:
Done. If we want to avoid type-casting here, we would have to type-cast it in `MonthsToPeriod` and `PeriodToMonths`. Also, we'll using these functions for `PERIOD_DIFF`, so this will avoid type-casting it there too.

@dbussink `mysql-server` uses `uint64` for these functions. Results would differ if we don't type-cast it into `uint64`.

Reply:
> @dbussink `mysql-server` uses `uint64` for these functions. Results would differ if we don't type-cast it into `uint64`. (https://github.com/mysql/mysql-server/blob/trunk/mysys/my_time.cc#L2254)

I don't think that's very relevant here since those are values way out of bounds anyway?

Reply:
But `months` can be negative here. So, we'll get different results for some negative cases e.g. `PERIOD_ADD(1206, -1231232332)`

Reply:
@dbussink We already run `FnPeriodAdd` in `cases.go` through some of these overflow cases (values in `inputBitwise` in `inputs.go`), which fail when we remove type-casting. Should we include more of these cases specifically in `FnPeriodAdd`?

Reply:
@dbussink I type casted it inside the function in the latest changes. That's why it's passing.

```go
func PeriodToMonths(period int64) int64 {
	p := uint64(period)
```

```go
func MonthsToPeriod(months int64) int64 {
	m := uint64(months)
```

The tests will fail, if this type casting is removed. I think I may have misunderstood. Did you mean something else?

Reply:
Ah yeah, that makes sense. I think we should cast it there where it makes most sense. If we also add `PERIOD_DIFF`, were should we put it you think?

Reply:
@dbussink I think it will be best to type cast inside the function (the way it's done now). That would eliminate the need to type cast separately. I can create a PR for `PERIOD_DIFF` once this is merged.